### PR TITLE
Update Virtuoso

### PR DESCRIPTION
**Pull request by Windvis**
This updates Virtuoso from 7.2.5.1 to 7.2.9 by switching to the redpencil/virtuoso:1.1.0 image.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,20 @@
...
+  - start the virtuoso container: `drc start virtuoso`
+  - create a new checkpoint so the transaction log is cleared:
+      - enter the isql-v environment: `drc exec virtuoso isql-v`
+      - create a checkpoint: `exec('checkpoint');`
+      - exit the isql-v environment: `exit;`
+  - stop the virtuoso service
+  - make a backup of the data/db folder in case something goes wrong and we need to downgrade again `cp -r data/db some-backup-folder`
+  - remove the transaction log file: `rm data/db/virtuoso.trx`
+  - `drc up -d virtuoso` to update to the new image
+  - check the logs to see if nothing strange happened
```
Comment (Windvis) [inline CHANGELOG.md:27]: Not sure if this is suited for the hosted environments, but this is what I did locally and it seemed to work.

**Comment by Windvis** [comment]
ah new version is out, so we should update to that one instead.

**Comment by cecemel** [comment]
We keep this one open until we have a sane version of virtuoso. I want this for documentation purpose.

**Comment by cecemel** [comment]
@Riadabd 
- the new version should be here: https://github.com/redpencilio/docker-virtuoso/pull/19
- to test in docker-compose.override.yml from dev-loket
-  @claire-lovisa has some deploy instructions to do it in a clean way, but no idea where.

**Comment by claire-lovisa** [comment]
Hello !

Before updating the version of virtuoso:
```
docker-compose exec database isql-v
    exec('checkpoint');
    exec('shutdown');

rm data/db/virtuoso.trx
```

Then you can bump the version and up the triplestore.

Note that I didn't do it for a while, so maybe try it out locally before doing it for real :)
You could also, if it's on prod, generate backups after doing the checkpoint, check the crontab of prod servers to see how the `virtuoso-backup.sh` script is called for your project

**Comment by Riadabd** [comment]
I'll put a couple of comments to document the process since it was buried inside Jira comments. The steps below involve the usual database update (some stuff will be edited after confirmation), and steps for nquad dumping and size saving will be posted in a separate comment.

### Steps for `Virtuoso`:

* Stop the stack: `docker compose down`
* Start the virtuoso container: `docker compose up -d virtuoso`
* Create a new checkpoint to clear the transaction log:
  - Enter the isql-v environment: `docker compose exec virtuoso isql-v`
  - Create a checkpoint: `exec('checkpoint');`
   - Exit the isql-v environment: `exec('shutdown');`
* Stop the virtuoso service: `docker compose down`
* Backup the data/db folder in case something goes wrong, and we need to downgrade again: `cp -r data/db some-backup-folder`
* Remove the transaction log file: `rm data/db/virtuoso.trx`
* Bump virtuoso to the new image
* `docker compose up -d virtuoso` to pick up the new version (currently `redpencil/virtuoso:1.2.0-rc.1`)
* Perform database workflows relating to your specific module
* Check the logs to monitor any irregularity

### Steps for `publication-triplestore`:

The steps are the same for publication-triplestore with some slight adjustments for the virtuoso log deletion:
* Stop the stack: `docker compose down`
* Start the publication-triplestore container: `docker compose up -d publication-triplestore`
* Create a new checkpoint to clear the transaction log:
  - Enter the isql-v environment: `docker compose exec publication-triplestore isql-v`
  - Create a checkpoint: `exec('checkpoint');`
  - Exit the isql-v environment: `exec('shutdown');`
* Stop the publication-triplestore service: `docker compose down`
* Backup the data/publication_triplestore folder in case something goes wrong, and we need to downgrade again: `cp -r data/publication-triplestore some-backup-folder`
* Remove the transaction log file: `rm data/publication-triplestore/virtuoso.trx`
* Bump publication-triplestore to the new image (currently `redpencil/virtuoso:1.2.0-rc.1`)
* `docker compose up -d publication-triplestore` to pick up the new version
* Perform workflows that go through publication-triplestore
* Check the logs to monitor any irregularity

**Comment by Riadabd** [comment]
# Upgrade without database size optimizations:

Before any upgrade starts, it's best to back up both `virtuoso` and `publication-triplestore`:
* Stop the stack: `docker compose down`
* Start the maintenance frontend banner
* Start the backup process using `virtuoso-backup.sh` inside `/data/useful-scripts/ or by following the below steps:
* `docker compose exec -i virtuoso_container mkdir -p backups` (`virtuoso_container` can be `virtuoso` or `publication-triplestore`)
* Place backups inside the `backups/` folder: `docker compose exec -i virtuoso_container isql-v`
  - `exec('checkpoint');`
  - `backup_context_clear();`
  - `backup_online('backup_',30000,0,vector('backups'));`
  - `exit;`

## Steps for `Virtuoso`:

* Stop the stack: `docker compose down` (The stack should already be down at this point if the backup step was performed)
* Start the virtuoso container: `docker compose up -d virtuoso`
* Create a new checkpoint to clear the transaction log:
  - Enter the isql-v environment: `docker compose exec virtuoso isql-v`
  - Create a checkpoint: `exec('checkpoint');`
   - Exit the isql-v environment: `exec('shutdown');`
* Stop the virtuoso service: `docker compose down`
* Remove the transaction log file: `rm data/db/virtuoso.trx`
* Bump virtuoso to the new image
* `docker compose up -d virtuoso` to pick up the new version (currently `redpencil/virtuoso:1.2.0-rc.1`)
* Perform database workflows relating to your specific module
* Check the logs to monitor any irregularity

## Steps for `publication-triplestore`:

The steps are the same for publication-triplestore with some slight adjustments for the virtuoso log deletion:
* Stop the stack: `docker compose down`
* Start the publication-triplestore container: `docker compose up -d publication-triplestore`
* Create a new checkpoint to clear the transaction log:
  - Enter the isql-v environment: `docker compose exec publication-triplestore isql-v`
  - Create a checkpoint: `exec('checkpoint');`
  - Exit the isql-v environment: `exec('shutdown');`
* Stop the publication-triplestore service: `docker compose down`
* Remove the transaction log file: `rm data/publication-triplestore/virtuoso.trx`
* Bump publication-triplestore to the new image (currently `redpencil/virtuoso:1.2.0-rc.1`)
* `docker compose up -d publication-triplestore` to pick up the new version
* Perform workflows that go through publication-triplestore
* Check the logs to monitor any irregularity

**Comment by Riadabd** [comment]
The upgrade process with and without applying the database size optimization process have been split for better clarity and separation.

**Pull request merged**